lib/src/main/java/com/cloudbees/groovy/cps/impl/AssignmentBlock.java

```diff
@@ -64,10 +64,10 @@ public Next fixLhs(Object lhs) {
         }
 
         /**
-         * Just straight assignment from RHS to LHS, then done
+         * Assign from RHS to LHS and pass RHS to the continuation.
          */
         public Next assignAndDone(Object rhs) {
-            return lhs.set(rhs,k);  // just straight assignment
+            return lhs.set(rhs, then(new ConstantBlock(rhs), e, k));
```

Comment:
Instead of passing the current continuation `k` to `LValue.set`, which would result in `k.receive(null)` being called after `lhs` is set to `rhs`, we construct a new continuation that calls `k.receive(rhs)` internally and pass it to `lhs.set` instead.

I think we could also add a new method like `setAndGet` to the `LValue` interface and implement it in the 2-3 subclasses of `LValue`, but this seemed simpler and I didn't see any other callers of `LValue.set` that would want the new behavior.